pkg/mimir/modules.go

```diff
@@ -542,23 +544,65 @@ func (t *Mimir) initRuler() (serv services.Service, err error) {
 
 	t.Cfg.Ruler.Ring.KVStore.Multi.ConfigProvider = multiClientRuntimeConfigChannel(t.RuntimeConfig)
 	t.Cfg.Ruler.Ring.ListenPort = t.Cfg.Server.GRPCListenPort
-	rulerRegisterer := prometheus.WrapRegistererWith(prometheus.Labels{"engine": "ruler"}, prometheus.DefaultRegisterer)
-	var queryable, federatedQueryable prom_storage.Queryable
-	// TODO: Consider wrapping logger to differentiate from querier module logger
-	queryable, _, eng := querier.New(t.Cfg.Querier, t.Overrides, t.Distributor, t.StoreQueryables, rulerRegisterer, util_log.Logger, t.ActivityTracker)
-
-	if t.Cfg.Ruler.TenantFederation.Enabled {
-		if !t.Cfg.TenantFederation.Enabled {
-			return nil, errors.New("-ruler.tenant-federation.enabled=true requires -tenant-federation.enabled=true")
+
+	var embeddedQueryable prom_storage.Queryable
+	var queryFunc rules.QueryFunc
+
+	if len(t.Cfg.Ruler.Querier.Address) > 0 {
```

Comment:
If it's a string, then this is clearer:

```suggestion
	if t.Cfg.Ruler.Querier.Address != "" {
```